Bug:下载ph站时发生错误

Console output
```
annie -d 'https://cn.pornhub.com/view_video.php?viewkey=ph5e35bb9dac481'

annie: version 0.9.8, A fast, simple and clean video downloader.


URL:         https://cn.pornhub.com/view_video.php?viewkey=ph5e35bb9dac481
Method:      GET
Headers:     http.Header{
    "User-Agent":      {"Mozilla/5.0 (Macintosh; Intel Mac OS X 10_13_6) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/69.0.3497.81 Safari/537.36"},
    "Referer":         {"https://cn.pornhub.com/view_video.php?viewkey=ph5e35bb9dac481"},
    "Accept":          {"text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8"},
    "Accept-Charset":  {"UTF-8,*;q=0.5"},
    "Accept-Encoding": {"gzip,deflate,sdch"},
    "Accept-Language": {"en-US,en;q=0.8"},
}
Status Code: 200
panic: runtime error: index out of range [0] with length 0

goroutine 1 [running]:
github.com/iawia002/annie/downloader.Download(0x555dfb478cae, 0x13, 0xc00048ad31, 0x2d, 0x555dfb4709d4, 0x5, 0xc0000a63f0, 0x0, 0x0, 0x0, ...)
        github.com/iawia002/annie/downloader/downloader.go:196 +0x180e
main.download(0x7ffcd4205260, 0x3d, 0x7ffcd4205260)
        github.com/iawia002/annie/main.go:184 +0x400
main.main()
        github.com/iawia002/annie/main.go:238 +0x184
```

--- a/extractors/pornhub/pornhub.go
+++ b/extractors/pornhub/pornhub.go
@@ -4,10 +4,13 @@ import (
 	"bytes"
 	"encoding/json"
 	"fmt"
+	"strings"
 
 	"github.com/iawia002/annie/extractors/types"
 	"github.com/iawia002/annie/request"
 	"github.com/iawia002/annie/utils"
+
+	"github.com/robertkrimen/otto"
 )
 
 type pornhubData struct {
@@ -48,9 +51,28 @@ func (e *extractor) Extract(url string, option types.Options) ([]*types.Data, er
 		return nil, err
 	}
 
+	scripts := utils.MatchAll(html, `<script type="text/javascript">(?s)(.+?)</script>`)
+	var flashvarJS string
+	for _, s := range scripts {
+		for _, js := range s[1:] {
+			if strings.Contains(js, "flashvars_") {
+				flashvarJS = js
+				break
+			}
+		}
+	}
+	vm := otto.New()
+	vm.Run(flashvarJS) // nolint
+	var urls []string
+	for i := 0; i < len(pornhubs); i++ {
+		if value, err := vm.Get(fmt.Sprintf("media_%d", i)); err == nil {
+			urls = append(urls, value.String())
+		}
+	}
+
 	streams := make(map[string]*types.Stream, len(pornhubs))
-	for _, data := range pornhubs {
-		if data.Format == "hls" {
+	for i, data := range pornhubs {
+		if data.Format != "mp4" {
 			continue
 		}
 
@@ -65,7 +87,7 @@ func (e *extractor) Extract(url string, option types.Options) ([]*types.Data, er
 		}
 		quality := string(data.Quality)
 
-		realURL := data.VideoURL
+		realURL := urls[i]
 		if len(realURL) == 0 {
 			continue
 		}

--- a/go.mod
+++ b/go.mod
@@ -10,9 +10,11 @@ require (
 	github.com/fatih/color v1.7.0
 	github.com/kr/pretty v0.1.0
 	github.com/mattn/go-colorable v0.0.9 // indirect
+	github.com/robertkrimen/otto v0.0.0-20191219234010-c382bd3c16ff
 	github.com/rylio/ytdl v0.6.2
 	github.com/tidwall/gjson v1.3.2
 	gopkg.in/cheggaaa/pb.v1 v1.0.28 // indirect
+	gopkg.in/sourcemap.v1 v1.0.5 // indirect
 )
 
 replace github.com/rylio/ytdl => github.com/mihaiav/ytdl v0.6.3-0.20200510100116-5f2bf8b4fec0

